app/models/article.rb

```diff
@@ -521,9 +521,12 @@ def validate_tag
     # check there are not too many tags
     return errors.add(:tag_list, "exceed the maximum of 4 tags") if tag_list.size > 4
 
-    # check tags names aren't too long
+    # check tags names aren't too long and don't contain non alphabet characters
     tag_list.each do |tag|
-      errors.add(:tag, "\"#{tag}\" is too long (maximum is 30 characters)") if tag.length > 30
+      new_tag = Tag.new(name: tag)
+      unless new_tag.valid?
```

Comment:
🔥

Reply:
I can take a look at the tests and make the changes

Reply:
I actually totally missed @maestromac's point bc I was confused when he said: "raises an error" which this doesnt do. What he means is that if we make a tag named "beginner" then if an Article tries to add that "beginning" tag here it will return as invalid bc Tag.new thinks we are trying to make a new "beginner" Tag.

One easy fix could be to ignore that uniqueness validation, however, that validation makes a db call which doing 4 times here for each tag is not great. I am trying to come up with a slick way to solve this problem but the easy fix is to duplicate the Regex for the `format` validation in the tag class. Maybe make a constant and use it in both places.

Reply:
I believe using a custom validation will do the trick. Let me push up what I have.